### PR TITLE
Fixes #396 #398:- Prototype of loklak media wall

### LINKED ISSUE #386: Adding option for "Display search results as media walls".
﻿Add option to "Display search results as media walls" on the top right of the feed page which would route to the media wall page. 

screen shot 2017-06-06 at 2 30 37 am

### LINKED ISSUE #396: New data structure and actions for the Queries in the application
**Parent Issue #395**

The Data Structure should be implemented to accommodate the attributes that define a query in general.
This must also be associated with the Actions which update the state represented by the data structure.
In general the structure must contain minimum of `queryString` and a `FilterList`.

### PR DESCRIPTION
**Pull request by Achint08**
**Changes proposed in this pull request**
- Added basic prototype of media wall
- Masonry view
- Added option to get different view  as a media wall

**TODO**
- Integrating media wall component with the search result (in the next PR).

**Screenshots (if appropriate)** 
screen shot 2017-06-21 at 6 22 47 pm

Media Wall View

screen shot 2017-06-21 at 6 22 23 pm

**Link to live demo (if appropriate):** 
https://achint08.github.io/loklak_search/
 

**Closes #386 , #398 **

Please review @mariobehling @hemantjadon @rishiraj824  @vibhcool @singhpratyush @djmgit @kavithaenair

### CODE REVIEW

File: src/app/media-wall/media-wall-card/media-wall-card.component.scss [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		span.img-span {
+			float: left;
+			position: relative;
+			width: 5.5vw;
+			height: 5.5vw;
+			border-width: 2px;
+			border-style: solid;
+			border-radius: 50%;
+			margin: 0 2.5vw 1vw 0;
+			overflow: hidden !important;
```
Comment (vibhcool) [inline src/app/media-wall/media-wall-card/media-wall-card.component.scss]: https://github.com/fossasia/loklak_search/pull/403#discussion_r123200326

**Comment by Achint08** [comment]
@vibhcool Done!

**Comment by Achint08** [comment]
@mariobehling Is this UI fine? Just have a look.

**Comment by mariobehling** [comment]
Thanks. It is a step in the right direction. If others accept it, it can be merged. Please ask your co-developers to review it.

Issues that need to be solved in follow up issues:
* Views text not aligned according to search button and it disappears when I click
* Media Walls text button not aligned. Please check how google.com does it.
* I don't see the same output of search results as on the normal view.

**Comment by Achint08** [comment]
@mariobehling Done!
> I don't see the same output of search results as on the normal view.
This is probably because  new API request is sent due to change in URL. This is in way correct, so that even if the query changes or user explicitly searches by URI, results are based on new query.

**Comment by mariobehling** [comment]
Thanks! 

Let's iterate from here.

**Pull request merged**